pkg/kubectl/cmd/apply.go

```diff
@@ -240,6 +240,17 @@ func RunApply(f cmdutil.Factory, cmd *cobra.Command, out, errOut io.Writer, opti
 			visitedNamespaces.Insert(info.Namespace)
 		}
 
+		// Add change-cause annotation to resource info if it should be recorded
+		if cmdutil.ShouldRecord(cmd, info) {
+			recordInObj := info.VersionedObject
```

Comment:
```
patchBytes, patchedObject, err := patcher.patch(info.Object, modified, info.Source, info.Namespace, info.Name)
```
It seems that patch is using `info.Object`. If storing the change in `info.VersionedObject`, it will cause the annotation missing when calculating the patch, right?